fcrepo-kernel-impl/src/test/java/org/fcrepo/integration/kernel/impl/FedoraResourceImplIT.java

```diff
@@ -452,6 +452,38 @@ public void testUpdateDenied() throws RepositoryException {
                         + " WHERE { }", new RdfStream());
     }
 
+    @Test (expected = IllegalArgumentException.class)
+    public void testInvalidSparqlUpdateValidation() throws RepositoryException {
+        final FedoraResource object =
+                containerService.findOrCreate(session, "/testRefObject");
+        object.updateProperties(
+                subjects,
+                "INSERT { <> <http://myurl.org/title/> \"fancy title\" . \n" +
+                " <> <http://myurl.org/title/> \"fancy title 2\" . } WHERE { }",
+                new RdfStream());
+    }
+
+    @Test
+    public void testValidSparqlUpdateValidation1() throws RepositoryException {
+        final FedoraResource object =
+                containerService.findOrCreate(session, "/testRefObject");
+        object.updateProperties(
+                subjects,
+                "INSERT { <> <http://myurl.org/title> \"5\" . } WHERE { }",
+                new RdfStream());
+    }
+
+    @Test
+    public void testValidSparqlUpdateValidation2() throws RepositoryException {
+        final FedoraResource object =
+                containerService.findOrCreate(session, "/testRefObject");
```

Comment:
Instead of a hardcoded String `/testRefObject`, please create the resource name with:
` final String pid = UUID.randomUUID().toString();`